### PR TITLE
[ADVPACK] files.c: Properly mark Wine diff

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Sort out code.

Addendum to 6eb8a1d (0.4.15-dev-6918).
See #5821.

## Proposed changes

- Properly mark Wine diff

### CODE REVIEW

File: dll/win32/advpack/files.c [hunk 1]
```diff
@@ -454,7 +458,7 @@ HRESULT WINAPI DelNodeW(LPCWSTR pszFileOrDirName, DWORD dwFlags)
     HRESULT ret = E_FAIL;
     
     TRACE("(%s, %d)\n", debugstr_w(pszFileOrDirName), dwFlags);
-
+
```
Comment (HBelusca) [inline dll/win32/advpack/files.c:461]: please revert

**Comment by HBelusca** [comment]
@whindsaks Could you review this? (it touches what you PR'ed)

File: dll/win32/advpack/files.c (hunk 1 above)
Comment (SergeGautherie) [reply dll/win32/advpack/files.c:461]: > revert

What is the purpose of this diff?

Comment (HBelusca) [reply dll/win32/advpack/files.c:461]: Removing extra trailing whitespace that has been added into this PR for unknown reasons

Comment (SergeGautherie) [reply dll/win32/advpack/files.c:461]: You got it wrong: my PR restores these whitespaces which have been removed for no reason.

**Comment by SergeGautherie** [comment]
Comments have been answered. Looking for a new review.

File: dll/win32/advpack/files.c [hunk 2]
```diff
@@ -341,17 +341,19 @@ static HRESULT DELNODE_recurse_dirtree(LPWSTR fname, DWORD flags)
...
             goto deleteinitialdirectory;
-        }
+#else
+        int fname_len = lstrlenW(fname);
 #endif
 
         /* Generate a path with wildcard suitable for iterating */
+#ifdef __REACTOS__
         fname_len = lstrlenW(fname);
+#endif
```
Comment (gonzoMD) [inline dll/win32/advpack/files.c:356]: I think this is less complicated and with fewer ReactOS guards (Lines 344-356):
```
        int fname_len = lstrlenW(fname);
#ifdef __REACTOS__
        if (flags & ADN_DEL_IF_EMPTY)
            goto deleteinitialdirectory;
#endif

        /* Generate a path with wildcard suitable for iterating */
```

**Review by gonzoMD — CHANGES_REQUESTED** [review]

File: dll/win32/advpack/files.c [hunk 3]
```diff
@@ -381,7 +383,9 @@ static HRESULT DELNODE_recurse_dirtree(LPWSTR fname, DWORD flags)
         {
 #ifdef __REACTOS__
 deleteinitialdirectory:
+#endif
             TRACE("%s: directory\n", debugstr_w(fname));
```
Comment (HBelusca) [inline dll/win32/advpack/files.c:387]: @SergeGautherie If this TRACE is indeed part of Wine code itself too, could you then change the diff as follows? (see next suggestion)

File: dll/win32/advpack/files.c [hunk 4]
```diff
@@ -381,7 +383,9 @@ static HRESULT DELNODE_recurse_dirtree(LPWSTR fname, DWORD flags)
...
+#ifdef __REACTOS__
```
Comment (HBelusca) [inline dll/win32/advpack/files.c:388]:
```suggestion
            TRACE("%s: directory\n", debugstr_w(fname));
#ifdef __REACTOS__
deleteinitialdirectory:
```
(This is a suggestion similar to what's already done in the next case for a file.)

File: dll/win32/advpack/files.c (hunk 2 above)
Comment (SergeGautherie) [reply dll/win32/advpack/files.c:356]: @whindsaks, Do you agree with this revert suggestion and explanation? As in, was your `lstrlenW()` change an optimization only?

File: dll/win32/advpack/files.c (hunk 4 above)
Comment (SergeGautherie) [reply dll/win32/advpack/files.c:388]: That would skip the `TRACE()`, in the `deleteinitialdirectory` case.
Do you confirm you want to change current behaviour?

File: dll/win32/advpack/files.c (hunk 2 above)
Comment (SergeGautherie) [reply dll/win32/advpack/files.c]: Reverted.

File: dll/win32/advpack/files.c (hunk 3 above)
Comment (SergeGautherie) [reply dll/win32/advpack/files.c:381]: It is from upstream.

File: dll/win32/advpack/files.c (hunk 4 above)
Comment (HBelusca) [reply dll/win32/advpack/files.c:382]: Ah yes, good catch @SergeGautherie . Let's keep it as it currently is in your PR.

**Review by HBelusca — APPROVED** [review]

**Review by JoachimHenze — APPROVED** [review]
Approved, although I do think the restoring of the end-of-line-whitespace is questionable. Better would be to submit a Wine-PR to strip that malformatting at the source.

**Comment by SergeGautherie** [comment]
Looking for a committer.

**Pull request merged**